larpandoracontent/LArControlFlow/MasterAlgorithm.cc

```diff
@@ -509,9 +509,24 @@ StatusCode MasterAlgorithm::RunSlicing(const VolumeIdToHitListMap &volumeIdToHit
 
 StatusCode MasterAlgorithm::RunSliceReconstruction(SliceVector &sliceVector, SliceHypotheses &nuSliceHypotheses, SliceHypotheses &crSliceHypotheses) const
 {
+    SliceVector selectedSliceVector;
+    if (m_shouldRunSlicing && m_sliceSelectionToolVector.size() > 0)
+    {
+        SliceVector inputSliceVector(sliceVector);
+        for (SliceSelectionBaseTool *const pSliceSelectionTool : m_sliceSelectionToolVector)
+        {
+            pSliceSelectionTool->SelectSlices(this, inputSliceVector, selectedSliceVector);
+            inputSliceVector = selectedSliceVector;
```

Comment:
Correct. I agree, this will probably never happen, albeit it does provide for combination purity and completeness cuts, so maybe tool one keeps all sufficiently complete slices, and then tool 2 trims it by purity to the purest of the sufficiently complete slices?